src/auth.js

```diff
@@ -233,17 +276,25 @@ function submitSignIn (url) {
     return Promise.resolve()
   }
 
-  return setUserInfo(url)
-    .then(() => {
-      var uI = document.getElementById('user-info')
-      if (uI) {
-        uI.innerHTML = getUserHTML()
-      }
+  // First, discover the authorized provider for a web id
```

Comment:
I presume this part of the code is incomplete because for WebIDs without an OIDC provider, it doesn't work (for both http and https WebIDs).

For http WebID it doesn't work because it has to go through a proxy, and the proxy doesn't forward the headers back to client - even if the response from the WebID has the openid-configuration in the header.

For https WebID there is no proxy involved but if that URLs response doesn't have a OIDC provider, it fails all the checks - this WebID be intended to for TLS.

We should continue to cover these cases. If proxy is going to be an issue, then maybe we skip checking OIDC for http WebID, and do what it is doing now - just fetch the profile. For https WebID, if can do OIDC checks first, if tails, fetch (GET) the profile.

We should aim to minimise the number of requests here.

What do you think?